zkevm-circuits/src/state_circuit/constraint_builder.rs

```diff
@@ -311,11 +412,28 @@ impl<F: Field> Queries<F> {
     fn address_change(&self) -> Expression<F> {
         self.address.value.clone() - self.address.value_prev.clone()
     }
+
+    fn tx_log_index(&self, is_pre: bool) -> Expression<F> {
+        if is_pre {
+            from_digits(&self.address.limbs_prev[0..2], (1u64 << 16).expr())
+        } else {
+            from_digits(&self.address.limbs[0..2], (1u64 << 16).expr())
+        }
+    }
+
+    fn tx_log_id(&self, is_pre: bool) -> Expression<F> {
+        if is_pre {
+            from_digits(&self.address.limbs_prev[2..4], (1u64 << 16).expr())
+        } else {
+            from_digits(&self.address.limbs[2..4], (1u64 << 16).expr())
+        }
+    }
```

Comment:
Have you considered turning these 2 methods into `tx_log_index`, `tx_log_index_prev`, `tx_log_id`, `tx_log_id_prev` for better readability?

Reply:
I prefer add new methods... For example, without IDE, readers see 'self. tx_log_id(true)' he will not know what it is.
But if he see 'self. tx_log_id_prev()', he know what it is just from the method name

Reply:
updated in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/552/commits/c3aa98efac241a5a9caf6a776d992244c5a80c13  thanks!